### PR TITLE
[fix][broker] Do not try to clean owned bundles from inactive source brokers (ExtensibleLoadManagerImpl only)

### PR DESCRIPTION
**Pull request by heesung-sn**
### Motivation

- Found a bug that the ownership monitor tries to clean owned bundles from inactive source brokers

### Modifications

- Fix the bug not to clean owned bundles from inactive source brokers
- Cleaned/refactored override ownership code
- Skip unloading LB internal topics. Instead, close those topics when shutting down the broker.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/heesung-sn/pulsar/pull/75

### CODE REVIEW

**Comment by dragosvictor** [comment]
Should this not be cherry-picked to 3.1 too?

**Comment by heesung-sn** [comment]
> Should this not be cherry-picked to 3.1 too?
Sure. I can push it there too.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java [hunk 1]
```diff
@@ -993,5 +997,26 @@ public void disableBroker() throws Exception {
...
+            futures.add(pulsar.getBrokerService().getTopicIfExists(name)
+                    .thenAccept(topicOptional -> topicOptional.ifPresent(topic -> topic.close(true)))
+                    .exceptionally(__ -> {
+                        log.warn("Failed to close internal topic:{}", name);
+                        return null;
+                    }));
+        }
+        try {
+            FutureUtil.waitForAll(futures)
+                    .get(pulsar.getConfiguration().getMetadataStoreOperationTimeoutSeconds(), TimeUnit.SECONDS);
```
Comment (dragosvictor) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java:1017]: It's not clear to me that the metadata store timeout is the most relevant choice here. What about `namespaceBundleUnloadingTimeoutMs` ?

Comment (heesung-sn) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java:1017]: good point. will update it.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 2]
```diff
@@ -2250,6 +2250,15 @@ private CompletableFuture<Integer> unloadServiceUnit(NamespaceBundle serviceUnit
         topics.forEach((name, topicFuture) -> {
             TopicName topicName = TopicName.get(name);
             if (serviceUnit.includes(topicName)) {
+                if (ExtensibleLoadManagerImpl.isLoadManagerExtensionEnabled(pulsar)
+                        && ExtensibleLoadManagerImpl.isInternalTopic(topicName.toString())) {
+                    if (log.isDebugEnabled()) {
```
Comment (Demogorgon314) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:2255]: Should we use `ExtensibleLoadManagerImpl.debug` here?

**Review by Demogorgon314 — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java (hunk 2 above)
Comment (heesung-sn) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java]: Agreed. I think it is better to put this log under `ExtensibleLoadManagerImpl.debug ` too.  Updated.

**Pull request merged**